### DISCUSSION TITLE
react-hook-form 7 setValue is not working

### DISCUSSION THREAD

**Discussion by mib-kamel** [react-hook-form/react-hook-form#6011] [Q&A] [answered]
Hello,

I am using react-hook-form v7,
When I try to use setValue I can't see any changes (no value changes)

I tried it inside setTimeout and directly and still doesn't work

`
const { register, unregister, handleSubmit, setValue, reset, formState: { isDirty, isValid, errors } } = useForm({ mode: 'onChange' });
`

` 
useEffect(() => {
            setTimeout(() => {
                setValue('firstName', 'Luo', {
                    shouldValidate: true
                })
            }, 2000)
    }, [setValue]);
` 

`
                        <div className="input-container">
                            <label htmlFor="firstName">First Name</label>
                            <Input type="text" name="firstName" placeholder="First Name" {...register("firstName", { required: true, minLength: 2, maxLength: 30 })} />
                        </div>
`

After a deep debugging I can see the firstName value changed successfully but it became unregistered.

Can you help me, please?

**Comment by bluebill1049** [comment, marked as answer]
provide a simple csb so we can take a look at it.

**Reply by mib-kamel** [reply]
Solved,

The problem was that I was using the "Input" of ant design,
when I used the normal "input" it got solved.

Thank you

**Reply by bluebill1049** [reply]
awesome!
